cisco_aci/datadog_checks/cisco_aci/fabric.py

```diff
@@ -228,6 +243,34 @@ def get_fabric_type(self, obj_type):
         if obj_type == 'l1PhysIf':
             return 'port'
 
+    def batch_payloads(self, devices, interfaces, collect_ts):
+        payloads = []
+
+        for device in devices:
+            payloads.append(
+                NetworkDevicesMetadata(namespace=self.namespace, devices=[device], collect_timestamp=collect_ts)
+            )
+
+        cur_resource_count = 0
+        cur_resources = []
+        for interface in interfaces:
+            if cur_resource_count == PAYLOAD_METADATA_BATCH_SIZE:
+                payloads.append(
+                    NetworkDevicesMetadata(
+                        namespace=self.namespace, interfaces=cur_resources, collect_timestamp=collect_ts
+                    )
+                )
+                cur_resources = []
+                cur_resource_count = 0
+            cur_resources.append(interface)
+            cur_resource_count += 1
+        if cur_resources:
+            payloads.append(
+                NetworkDevicesMetadata(namespace=self.namespace, interfaces=cur_resources, collect_timestamp=collect_ts)
+            )
+
+        return payloads
```

Comment:
Pro-tip (but not a suggested change), you can leverage the power of `yield` in such cases:

```python
    def batch_payloads(self, devices, interfaces, collect_ts):

        for device in devices:
            yield NetworkDevicesMetadata(namespace=self.namespace, devices=[device], collect_timestamp=collect_ts)

        payloads = []
        for interface in interfaces:
            if len(payloads) == PAYLOAD_METADATA_BATCH_SIZE:
                yield NetworkDevicesMetadata(
                    namespace=self.namespace, interfaces=cur_resources, collect_timestamp=collect_ts
                )
                payloads = []
            payloads.append(interface)
        if payloads:
            yield NetworkDevicesMetadata(namespace=self.namespace, interfaces=cur_resources, collect_timestamp=collect_ts))
```